### ISSUE TITLE
Scroll down option for selecting months and years

### ISSUE THREAD

**Issue by 07tAnYa** [plone/volto#5662]
**Is your feature request related to a problem? Please describe.**
When I go to select a date for my event, I have to keep on right/left swiping to select the appropriate date.

**Describe the solution you'd like**
When I hover on the year, it should show the years in a scroll-down-like structure, so that users can select easily.

**Additional context**
Here is the screen recording of the above-discussed issue: 

https://github.com/plone/volto/assets/140234197/5f157d0f-7ad3-44ad-ad33-ffc8f9be2053

**Comment by stevepiercy** [comment]
@07tAnYa we use the datetime picker as shown in https://6.docs.plone.org/storybook/?path=/story/edit-widgets-datetime--datetime.

However, there may be better datetime pickers available. I brought this up for discussion in a PLIP at https://github.com/plone/volto/issues/5629. I agree that clicking through it, with limited keyboard shortcuts, is bad UX and accessibility. If you have an idea for a better off-the-shelf widget, please add your comment to that issue. Thank you!

**Comment by 07tAnYa** [comment]
@stevepiercy Sir, so should I close this issue over here?

**Comment by stevepiercy** [comment]
No.

> If you have an idea for a better off-the-shelf widget, please add your comment to that issue.

**Comment by vr-varad** [comment]
@07tAnYa is this issue still open

**Comment by stevepiercy** [comment]
@vr-varad please read and follow [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do) item 4.

**Comment by vikash-prem123** [comment]
can i change the date widget like this so it will be easy to select.

![Screenshot from 2024-02-05 10-15-36](https://github.com/plone/volto/assets/106796672/dfc3f63a-1f43-41a2-a5bb-5074bbf2e489)

**Comment by davisagli** [comment]
@vikash-prem123 No, that will hurt usability in other ways. We'd like to keep the existing calendar, but add a way to jump to a different month/year.

**Comment by vikash-prem123** [comment]
Okay i have an idea. So when i click on the year ( on top )  and then  left arrow or right arrow  will change the year and click back it will be normal (or change the months).
By default it will change months.

![Screenshot from 2024-02-05 10-23-23](https://github.com/plone/volto/assets/106796672/49b51036-a9dc-4db5-848c-0fc32e14dcac)

**Comment by stevepiercy** [comment]
@vikash-prem123 before you go reinventing the wheel, first determine what we currently use for a date picker, and see if that package or library can be modified with a configuration option. Include any references to its documentation.

**Comment by 07tAnYa** [comment]
> @vikash-prem123 before you go reinventing the wheel, first determine what we currently use for a date picker, and see if that package or library can be modified with a configuration option. Include any references to its documentation.

In the volto interface of plone 6, we are currently using singledatepicker a custom component of react-dates.
Because of its light weight it's in the use, but for more friendliness like for support of single click on year or month, it's complete content should be shown so that the users can select the desired date easily.
The conversations we had earlier on this topic, ended with that the team hasn't decided yet which format should be displayed to the users, so because of this in priority we are now more focused on whether to switch to some more advanced features like momentjs or to upgrade the python libraries only for this .

@stevepiercy your insights on this would be highly appreciated.
And for the documentation of dates , I found plone 3 documentation to be more useful for this. Like how selecting dates and everything related to this works. 
Here's the link:  https://3.docs.plone.org/working-with-content/using-collections/oldstyle/using-and-understanding-dates.html

**Comment by vikash-prem123** [comment]
@stevepiercy and @07tAnYa, thank you for your input and suggestions.

I will investigate the current date picker implementation and check if it can be modified to include the desired configuration option for changing between year and month views. I'll look into the documentation and see if there are any existing features or options that can help us achieve this.

@stevepiercy, if you have any insights or recommendations regarding the date picker or the use of advanced features like momentjs, your input would be valuable.

I'll check the provided documentation link for additional insights into working with dates.

**Comment by stevepiercy** [comment]
@vikashsprem @07tAnYa we plan to implement https://codesandbox.io/p/sandbox/affectionate-rosalind-tdm323?file=%2Fsrc%2FCalendar.js

**Comment by 07tAnYa** [comment]
> @vikashsprem @07tAnYa we plan to implement https://codesandbox.io/p/sandbox/affectionate-rosalind-tdm323?file=%2Fsrc%2FCalendar.js

This is really good.

**Comment by PriyanshuG24** [comment]
Hi, I noticed this issue is labeled as an enhancement, and it seems like a valuable improvement to the project. I would love to contribute by working on this. Is anyone currently working on it, or would it be okay if I start working on it?

Thank you!

**Comment by stevepiercy** [comment]
@PriyanshuG24 please read and follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do), [Contributing to Plone](https://6.docs.plone.org/contributing/index.html), and [Contributing to Volto](https://6.docs.plone.org/contributing/volto.html).

**Comment by Manas-Kenge** [comment]
@stevepiercy I am now solving the issue as per  https://codesandbox.io/p/sandbox/affectionate-rosalind-tdm323?file=%2Fsrc%2FCalendar.js. The problem I am facing is with react-aria package. Is it ok to install it or am I supposed to import the components from somewhere else?

**Comment by stevepiercy** [comment]
@Manas-Kenge sorry, I don't know. I do know that there is a plan to use React Aria many of the components, and away from Semantic UI. I think this issue depends on the work being done in https://github.com/plone/volto/issues/4352. Hopefully someone else can provide direction about implementation details. We have another meeting in a couple of weeks to review priorities for Plone 7, and I would like to see this one prioritized. Until then, I wouldn't start work on it.
